tools/protoc-plugin/src/main/java/io/spine/tools/protoc/validation/ValidatorGenerator.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright 2019, TeamDev. All rights reserved.
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.tools.protoc.validation;
+
+import com.google.common.collect.ImmutableSet;
+import io.spine.code.java.ClassName;
+import io.spine.protobuf.MessageWithConstraints;
+import io.spine.tools.protoc.CompilerOutput;
+import io.spine.tools.protoc.InsertionPoint;
+import io.spine.tools.protoc.NoOpGenerator;
+import io.spine.tools.protoc.ProtocPluginFiles;
+import io.spine.tools.protoc.SpineProtoGenerator;
+import io.spine.tools.protoc.SpineProtocConfig;
+import io.spine.tools.protoc.iface.MessageImplements;
+import io.spine.tools.protoc.iface.MessageInterface;
+import io.spine.tools.protoc.iface.PredefinedInterface;
+import io.spine.tools.validate.MessageValidatorFactory;
+import io.spine.type.MessageType;
+import io.spine.type.Type;
+
+import java.util.Collection;
+
+import static io.spine.tools.protoc.InsertionPoint.builder_scope;
+import static io.spine.tools.protoc.InsertionPoint.class_scope;
+import static io.spine.tools.protoc.TypeParameters.empty;
+import static io.spine.tools.protoc.iface.MessageImplements.implementInterface;
+
+/**
+ * The message validation code generator.
+ *
+ * <p>Generates code which validates message fields upon the constraints, as well as the API which
+ * exposes validation to the user of the message class.
+ */
+public final class ValidatorGenerator extends SpineProtoGenerator {
```

Comment:
I think we need to re-visit the name `SpineProtoGenerator` (including its description). Now it reads as if we were generating `.proto` files, which is not the case.

`SpineCodeGenerator` or simply `CodeGenerator` would work fine, assuming we mention that we do take `.proto` code as the input.

Reply:
While the prefix `Spine` in `SpineProtocConfig` serves for the purpose of saying that the configuration is specific to the Spine framework, having the same prefix for the code generation classes looks excessive. We already have this notion in the package name, and we don't really need to highlight the fact again (as for `SpineProtocConfig`).

Reply:
In `SpineProtocConfig` the name points at the fact that the configurated Protoc plugin is called `spineProtoc`.
For `SpineProtoGenerator`, I accept your suggestion.